crystal_diffusion/models/egnn_utils.py

```diff
@@ -85,3 +87,33 @@ def get_edges_batch(n_nodes: int, batch_size: int) -> torch.Tensor:
             all_edges.append(edges + n_nodes * i)
         edges = torch.cat(all_edges)
     return edges
+
+
+def get_edges_with_radial_cutoff(relative_coordinates: torch.Tensor, unit_cell: torch.Tensor,
+                                 radial_cutoff: float = 4.0, drop_duplicate_edges: bool = True) -> torch.Tensor:
+    """Get edges for a batch with a cutoff based on distance.
+
+    Args:
+        relative_coordinates: batch x n_atom x spatial dimension tensor with relative coordinates
+        unit_cell: batch x spatial dimension x spatial dimension tensor with the unit cell vectors
+        radial_cutoff (optional): cutoff distance in Angstrom. Defaults to 4.0
+        drop_duplicate_edges (optional): if True, return only 1 instance of each edge. If False, return each edge
+            multiple times, depending on the unit cell shift multiplicities. Defaults to True.
+
+    Returns:
+        long tensor of size [number of edges, 2] with edge indices
+    """
+    cartesian_coordinates = relative_coordinates @ unit_cell  # convert back to cartesian coordinates
```

Comment:
We have a method to do this: 
`crystal_diffusion.utils.basis_transformations.get_positions_from_coordinates`
I suggest using this method as much as possible, because it will remove the issue of confusing the "basis_vectors" as column vectors or row vectors. This is not an issue when the unit cell is orthogonal, but it could introduce subtle bugs if we go on to non-orthogonal unit cells...